### PR TITLE
[spv-in] Fix radians/degrees parsing

### PR DESCRIPTION
**Pull request by expenses**
See https://github.com/gfx-rs/naga/pull/797#issuecomment-829284308.

### CODE REVIEW

File: src/front/spv/mod.rs
Comment (kvark) [inline src/front/spv/mod.rs:1743]: Let's freeze this until #817 lands (and then re-implement on top of it). After it, you'd just call `self.get_constant_scalar(..)` here, and you could remove `maths_constant_expressions` entirely

**Review by kvark — CHANGES_REQUESTED** [review]

File: src/front/spv/mod.rs
Comment (kvark) [reply src/front/spv/mod.rs:1743]: this is merged now

Comment (expenses) [reply src/front/spv/mod.rs:1743]: I can only see a `get_constant_scalar` function in `back/spv`..

Comment (kvark) [reply src/front/spv/mod.rs:1743]: oh, sorry, I confused myself

File: src/front/spv/mod.rs
Comment (kvark) [inline src/front/spv/mod.rs:2243]: given that this is a very narrow use case, can we get away with 1 constant instead of 2?

File: src/front/spv/mod.rs
Comment (kvark) [inline src/front/spv/mod.rs:430]: This is very similar to index constants, in spirit: in both cases we keep a hold of a constant handle, and in both cases there is an associated expression handle. We should probably merge them.

**Comment by expenses** [comment]
Sorry for having this on-hold for so long! I'll try and get it done today.

File: src/front/spv/mod.rs
Comment (expenses) [reply src/front/spv/mod.rs:3364]: It's not a huge deal, but IMO the resulting shaders are going to be slightly nicer if we use 2 constants and multiply by both, rather than having 1 and choosing to multiply or divide.

File: src/front/spv/mod.rs
Comment (expenses) [reply src/front/spv/mod.rs:545]: We could merge them, but this would make things a lot messier. The `index_constant_expressions` are currently indexed by the index of the vector component that is being extracted. Either we change that by an offset, or we change `DEGREES_TO_RADIANS_CONST_INDEX` and `RADIANS_TO_DEGREES_CONST_INDEX` to go after these 4 index constants. I'd prefer to keep things separate if possible.

**Comment by kvark** [comment]
Needs a rebase

**Comment by kvark** [comment]
Sorry, would you mind rebasing again?

**Comment by expenses** [comment]
To be honestly, I'm not entirely sure why pull requests need to be rebased. As I'm not currently involved in naga or using wgpu, I'd rather you take over the PR and merge it.

**Comment by kvark** [comment]
We can't merge it while there are merge conflicts, hence my request.

**Comment by kvark** [comment]
I rewrote this to avoid any pre-work done, and added a test.

**Review by kvark — APPROVED** [review]

**Pull request merged**
